ledger-tool/src/main.rs

```diff
@@ -2263,6 +2264,18 @@ fn main() {
                 let starting_slot = value_t_or_exit!(arg_matches, "starting_slot", Slot);
                 let ending_slot = value_t_or_exit!(arg_matches, "ending_slot", Slot);
                 let target_db = PathBuf::from(value_t_or_exit!(arg_matches, "target_db", String));
+                let target_shred_storage_type = match ShredStorageType::from_ledger_path(
+                    &target_db,
+                    DEFAULT_LEDGER_TOOL_ROCKS_FIFO_SHRED_STORAGE_SIZE_BYTES,
+                ) {
+                    Some(s) => s,
+                    None => {
+                        error!("Shred storage type of target_db cannot be inferred, the default RocksLevel will be used");
+                        error!("To use fifo shred storage type, create {} folder under the specified target_db directory",
+                                ShredStorageType::RocksFifo(BlockstoreRocksFifoOptions::default()).blockstore_directory());
+                        ShredStorageType::RocksLevel
+                    }
+                };
```

Comment:
Actually two things:
1) We have almost this exact logic already; can you move this into a little helper function? I know this comment mentions `target_db`; however, we could modify the error string to include the path incase the user hits this error and needs to be able to distinguish 
> Shred storage type of blockstore at {} cannot be inferred, the default RockLevel will be used

Also, given that we'll just use `RocksLevel`, I think `warn!` seems more appropriate for log level

2) I'm unsure if the second error message is appropriate here. There are two cases where we'll enter the `None` case:
- There are no directories for either compaction type (ie copy would create brand new ledger)
- There are directories for both compaction types

In the first case, we would be giving instructions but do the command anyways. If the second case, that folder already exists so this is kind of confusing